commit a4f9453542385d8e5e40dd71c00cc0d3a8a3a12e
Author: Tobias Wolf <dev-NTEO@vplace.de>
Date:   Wed Nov 23 10:40:07 2016 +0100

    of: Add check to of_scan_flat_dt() before accessing initial_boot_params
    
    commit 3ec754410cb3e931a6c4920b1a150f21a94a2bf4 upstream.
    
    An empty __dtb_start to __dtb_end section might result in
    initial_boot_params being null for arch/mips/ralink. This showed that the
    boot process hangs indefinitely in of_scan_flat_dt().
    
    Signed-off-by: Tobias Wolf <dev-NTEO@vplace.de>
    Cc: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
    Cc: linux-mips@linux-mips.org
    Patchwork: https://patchwork.linux-mips.org/patch/14605/
    Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
    Signed-off-by: Amit Pundir <amit.pundir@linaro.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/of/fdt.c b/drivers/of/fdt.c
index 6a43fd3..502f554 100644
--- a/drivers/of/fdt.c
+++ b/drivers/of/fdt.c
@@ -741,9 +741,12 @@ int __init of_scan_flat_dt(int (*it)(unsigned long node,
 	const char *pathp;
 	int offset, rc = 0, depth = -1;
 
-        for (offset = fdt_next_node(blob, -1, &depth);
-             offset >= 0 && depth >= 0 && !rc;
-             offset = fdt_next_node(blob, offset, &depth)) {
+	if (!blob)
+		return 0;
+
+	for (offset = fdt_next_node(blob, -1, &depth);
+	     offset >= 0 && depth >= 0 && !rc;
+	     offset = fdt_next_node(blob, offset, &depth)) {
 
 		pathp = fdt_get_name(blob, offset, NULL);
 		if (*pathp == '/')
